src/test/java/org/opensearch/security/auth/ldap/LdapBackendTest.java

```diff
@@ -116,7 +117,27 @@ public void testLdapAuthenticationBindDn() throws Exception {
             .put(ConfigConstants.LDAP_AUTHC_USERSEARCH, "(uid={0})")
             .put(ConfigConstants.LDAP_AUTHC_USERBASE, "ou=people,o=TEST")
             .put(ConfigConstants.LDAP_BIND_DN, "cn=Captain Spock,ou=people,o=TEST")
-            .put(ConfigConstants.LDAP_PASSWORD, "spocksecret")
+            .put(ConfigConstants.LDAP_PASSWORD.insecurePropertyName, "spocksecret")
+            .build();
+
+        final LdapUser user = (LdapUser) new LDAPAuthenticationBackend(settings, null).authenticate(
+            new AuthCredentials("jacksonm", "secret".getBytes(StandardCharsets.UTF_8))
+        );
+        Assert.assertNotNull(user);
+        assertThat(user.getName(), is("cn=Michael Jackson,ou=people,o=TEST"));
+    }
+
+    @Test
+    public void testLdapAuthenticationBindDnWithSecurePassword() throws Exception {
+        final var mockSecureSettings = new MockSecureSettings();
+        mockSecureSettings.setString(ConfigConstants.LDAP_PASSWORD.propertyName, "spocksecret");
+
+        final Settings settings = Settings.builder()
+            .putList(ConfigConstants.LDAP_HOSTS, "localhost:" + ldapPort)
+            .put(ConfigConstants.LDAP_AUTHC_USERSEARCH, "(uid={0})")
+            .put(ConfigConstants.LDAP_AUTHC_USERBASE, "ou=people,o=TEST")
+            .put(ConfigConstants.LDAP_BIND_DN, "cn=Captain Spock,ou=people,o=TEST")
+            .setSecureSettings(mockSecureSettings)
```

Comment:
I think we need one integration test for secure settings. You can find integration tests for LDAP here:

https://github.com/opensearch-project/security/blob/main/src/integrationTest/java/org/opensearch/security/http/LdapAuthenticationTest.java